plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/expression/ControlFlow.kt

```diff
@@ -207,7 +207,8 @@ data class FunctionExp(val signature: FullNamedFunctionSignature?, val body: Exp
             // Ideally we would want to assume these rather than inhale them to prevent inconsistencies with permissions.
             // Unfortunately Silicon for some reason does not allow Assumes. However, it doesn't matter as long as the
             // provenInvariants don't contain permissions.
-            arg.provenInvariants().forEach { invariant ->
+            // TODO (inhale vs require) Decide if `predicateAccessInvariant` should be required rather than inhaled in the beginning of the body.
+            (arg.provenInvariants() + listOfNotNull(arg.predicateAccessInvariant())).forEach { invariant ->
```

Comment:
Is this change necessary for this PR to work?  If not, I would recommend not including it.

Reply:
Without this change I can't add any tests where function call accepts some subtype of the type for which it was defined, only the type itself. I thought it was important to show that implementation of the properties works in that case.

Reply:
Is that because more work is needed on unfolding in that case?

Reply:
You kind of need to unfold the predicate until the necessary type is met.